json_options.go

```diff
@@ -0,0 +1,88 @@
+package zap
+
+import "time"
+
+// JSONOption is used to set options for a JSON encoder.
+type JSONOption interface {
```

Comment:
Unexporting `Apply` definitely makes sense...not sure why I exported it to begin with. We should keep `JSONOption` exported, though, since it'll be part of the public `NewJSONEncoder` signature.